src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -386,13 +386,12 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_delete_node(
 
 std::shared_ptr<AbstractOperator> LQPTranslator::_translate_update_node(
     const std::shared_ptr<AbstractLQPNode>& node) const {
-  const auto input_operator = translate_node(node->left_input());
   auto update_node = std::dynamic_pointer_cast<UpdateNode>(node);
 
-  auto new_value_exprs = _translate_expressions(update_node->node_expressions, node);
+  const auto input_operator_left = translate_node(node->left_input());
```

Comment:
(random line)

```
CREATE TABLE t (i INT, l LONG, f FLOAT, d DOUBLE)
INSERT INTO t (i,l,f,d) VALUES (1,2,3,4);
UPDATE t SET i = 1, f = 1, l = 1
libc++abi.dylib: terminating with uncaught exception of type std::logic_error: /Users/markus/Projekte/Opossum/Git/src/lib/storage/value_segment.cpp:124 This ValueSegment does not support null values.
```

Seems to have something to do with float values.

Maybe have a look if you want to merge in the current sql-parser and mrks/fixdouble, where I tried to fix doubles being treated as floats.

Reply:
Couldn't we add a bogus `WHERE 1` for that case?

Reply:
Yes, I guess I was on the wrong branch

Reply:
Ah, wait. We can be sure that we have a reference table because we are guaranteed to have a validate before. We can't run update on non-MVCC tables.

Reply:
If anything, this should be a `DebugAssert(lqp_is_validated(...), ...);`

Reply:
I didn't think unconditional UPDATES had a practical use and thought we didn't want to support them because they are more dangerous than useful. Do you want me to add proper support for unconditional UPDATEs as part of this PR?

Reply:
I thought we already had that support before? What is missing?

Think of queries like `UPDATE employees SET salary = salary * 1.2` (I wish...)